dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/hdfs/ScanContext.java

```diff
@@ -79,23 +100,38 @@ void endWalkDir(FileStatus dir, long nFiles, long nDirs, long accumFileSize) thr
       this.numDirs += nDirs;
       this.accumulatedFileSize += accumFileSize;
 
+      String absolutePath = dir.getPath().toUri().getPath();
+      HdfsFileStatus hdfsFileStatus = dfsClient.getFileInfo(absolutePath);
```

Comment:
It is indeed a potentially slow network op (when running on the cluster it takes < 1ms but when running remotely it takes a long time).
What do you mean by hoisted -- rewrite it (and the following code) as another Task/Job submitted to the same ExecutorManager? (or something else)

Reply:
Got it!
Thanks for pointing that out, it did not occur to me at all.

Reply:
And there was a substantial speed up after fixing this ...

Reply:
That's the one. Nicely done, thank you. It probably didn't show up when you were "very local" to your HDFS cluster, but when the network latency went up, it probably killed the performance. You can spot that in visualvm's threads view, because all the threads are orange except one.